aws/retry/middleware_test.go

```diff
@@ -252,6 +299,66 @@ func TestAttemptMiddleware(t *testing.T) {
 				},
 			},
 			Err: fmt.Errorf("some error"),
+			ExpectResults: AttemptResults{Results: []AttemptResult{
+				{
+					Err: fmt.Errorf("some error"),
+				},
+			}},
+		},
+		"nested metadata and valid response": {
+			Next: func(retries *[]retryMetadata) middleware.FinalizeHandler {
+				num := 0
+				reqsErrs := []error{
+					mockRetryableError{b: true},
+					nil,
+				}
+				return middleware.FinalizeHandlerFunc(func(ctx context.Context, in middleware.FinalizeInput) (out middleware.FinalizeOutput, metadata middleware.Metadata, err error) {
+					m, ok := getRetryMetadata(ctx)
+					if ok {
+						*retries = append(*retries, m)
+					}
+					if num >= len(reqsErrs) {
+						err = fmt.Errorf("more requests then expected")
+					} else {
+						err = reqsErrs[num]
+						num++
+					}
+
+					if err != nil {
+						metadata.Set("testKey", "testValue")
+					} else {
+						out.Result = "mockResponse"
```

Comment:
`out.Result` will be the deserialized output not the HTTP response right?  I think the `AttemptResult.Response` should be the deserialized transport (e.g. HTTP) raw response message.  This would be needed in order to provide access to the headers of the raw response.